java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/ArrowDatabaseMetadata.java

```diff
@@ -754,7 +754,11 @@ private <T> T getSqlInfoAndCacheIfCacheIsEmpty(final SqlInfo sqlInfoCommand,
   }
 
   private String convertListSqlInfoToString(final List<?> sqlInfoList) {
```

Comment:
I think it would make more sense to return `Optional<String>` and update client code to explicitly decide what they want to do with missing values.

Reply:
This method is used by these methods:
* getSqlKeywords()
* getNumericFunctions()
* getStringFunctions()
* getSystemFunctions()
* getTimeDateFunctions()

These methods come from Avatica, so we can't easily change the return type.

https://javadoc.io/static/org.apache.calcite.avatica/avatica-core/1.24.0/org/apache/calcite/avatica/AvaticaDatabaseMetaData.html#getSQLKeywords--

Reply:
But we can update our implementations to be explicit, right? 

In any case, defaulting to an empty string also seems fine so long as it's documented in the helper's docstring

Reply:
Updated the return type of `convertListSqlInfoToString()` to a `Optional<String>`.